contracts/Kasu.sol

```diff
@@ -92,17 +93,41 @@ contract Kasu is KasuStorage {
 
     // [Feature 3] Borrower's dashboard
     // borrower can see all the NFTs they borrowed
-    function viewRentedListings(uint256 listingId) public {
-
+    function viewRentedListings(address borrowerAddress) public {
+        // Iterating with a guard is expesnive...
     }
 
     // [Feature 3] Borrower's dashboard
     // After borrower return NFT, collateral is sent from smart contract to borrower's address
     function returnNFT(uint256 listingId) public {
-
+        Listing memory listing = _getListingById(listingId);
+        require(
+            _assertOwnership(
+                listing.tokenAddress,
+                listing.lenderAddress,
+                listing.tokenId
+            ) == true,
+            "Token not returned yet"
+        );
+        // TODO verify the property houses the collateral amount, what about interest?
+        require(
+            address(this).balance <= listing.collateralRequired,
+            "Insufficient ether in contractAddress"
+        );
+        listing.lenderAddress.transfer(listing.collateralRequired);
```

Comment:
collateral should go to the borrower on the normal return path. Interest goes to the lender.